spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalog.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.catalog.TableChange;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+/**
+ * GravitinoCatalog is the class registered to Spark CatalogManager, it's lazy loaded which means
+ * Spark connector loads GravitinoCatalog when it's used. It will create different Spark Table for
+ * different Gravitino catalog.
+ */
+public class GravitinoCatalog implements TableCatalog, SupportsNamespaces {
+  protected TableCatalog sparkCatalog;
+  protected Catalog gravitinoCatalog;
+
+  private String metalakeName;
+  private String catalogName;
+  private GravitinoCatalogManager gravitinoCatalogManager;
+
+  public GravitinoCatalog() {
+    gravitinoCatalogManager = GravitinoCatalogManager.get();
+    metalakeName = gravitinoCatalogManager.getMetalakeName();
+  }
+
+  @Override
+  public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
+    throw new NotSupportedException("Doesn't support listing table");
+  }
+
+  @Override
+  public Table createTable(
+      Identifier ident, Column[] columns, Transform[] partitions, Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    throw new NotSupportedException("Doesn't support creating table");
+  }
+
+  // Will create a catalog specific table by invoking createSparkTable()
+  @Override
+  public Table loadTable(Identifier ident) throws NoSuchTableException {
+    throw new NotSupportedException("Doesn't support loading table");
+  }
+
+  @SuppressWarnings("deprecation")
+  @Override
+  public Table createTable(
+      Identifier ident, StructType schema, Transform[] partitions, Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    throw new NotSupportedException("Deprecated create table method");
+  }
+
+  @Override
+  public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchTableException {
+    throw new NotSupportedException("Doesn't support altering table");
+  }
+
+  @Override
+  public boolean dropTable(Identifier ident) {
+    throw new NotSupportedException("Doesn't support drop table");
+  }
+
+  @Override
+  public void renameTable(Identifier oldIdent, Identifier newIdent)
+      throws NoSuchTableException, TableAlreadyExistsException {
+    throw new NotSupportedException("Doesn't support renaming table");
+  }
+
+  @Override
+  public void initialize(String name, CaseInsensitiveStringMap options) {
+    this.catalogName = name;
+    this.gravitinoCatalog = gravitinoCatalogManager.getGravitinoCatalogInfo(name);
+    String provider = gravitinoCatalog.provider();
+    Preconditions.checkArgument(provider != null, name + " catalog provider is null");
+    this.sparkCatalog = createAndInitSparkCatalog(provider, name, options);
```

Comment:
Why do we need a spark catalog, which is a hive catalog here, and also have a Gravitino catalog here?

Reply:
the name of GravitinoCatalog is confusing, one is the catalog name registered to Spark catalog manager, the other is the actually a GravitinoCatalogClient to do schema operations. I rename it to `GravitinoCatalogClient`.
sparkcatalog is the specific catalog to do IO operations.